tests/unit/test-class-wc-payments-utils.php

```diff
@@ -316,4 +316,22 @@ public function test_redact_array_handles_non_arrays() {
 
 		$this->assertEquals( $expected, $result );
 	}
+
+	public function test_get_order_original_currency() {
+		$order             = WC_Helper_Order::create_order();
+		$original_currency = $order->get_currency();
+
+		// make sure WC_Payments::maybe_set_original_order_currency() sets _wcpay_original_order_currency.
+		$this->assertEquals( $order->get_meta( '_wcpay_original_order_currency' ), $original_currency );
```

Comment:
`_wcpay_original_order_currency` -> `_wcpay_stripe_intent_currency` or something similar, if it has to remain ofcourse =)

Reply:
Let's avoid Stripe here, `_wcpay_intent_currency` or `_wcpay_customer_currency` should be good enough.

Reply:
`_wcpay_intent_currency` it is!